packages/react/src/generators/federate-module/lib/utils.ts

```diff
@@ -0,0 +1,268 @@
+import {
+  ChangeType,
+  ProjectGraph,
+  StringChange,
+  Tree,
+  createProjectGraphAsync,
+  joinPathFragments,
+  readCachedProjectGraph,
+} from '@nx/devkit';
+import { join } from 'path';
+import { ensureTypescript } from '@nx/js/src/utils/typescript/ensure-typescript';
+import * as ts from 'typescript';
+
+let tsModule: typeof import('typescript');
+
+if (!tsModule) {
+  tsModule = ensureTypescript();
+}
+
+/**
+ * Adds a Module Federation path to the exposes property of the module federation config
+ * The assumption here is made the we will only update a TypeScript Module Federation file namely 'module-federation.config.ts'
+ * @param tree Tree for the workspace
+ * @param projectPath Project path relative to the workspace
+ * @param moduleName The name of the module to expose
+ * @param modulePath The path to the module to expose relative to the remote source root (e.g. './src/my-lib/my-lib.ts')
+ */
+export function addPathToExposes(
+  tree: Tree,
+  projectPath: string,
+  moduleName: string,
+  modulePath: string
+) {
+  const moduleFederationConfigPath = joinPathFragments(
+    projectPath,
+    'module-federation.config.ts'
+  );
+  const configCode = tree.read(moduleFederationConfigPath).toString();
+  const source = tsModule.createSourceFile(
+    moduleFederationConfigPath,
+    configCode,
+    tsModule.ScriptTarget.Latest,
+    true
+  );
+
+  // Create a printer to print the updated AST
+  const printer = tsModule.createPrinter();
+
+  const updatedSourceCode = tsModule.transform(source, [
+    transformer(moduleName, modulePath),
+  ]).transformed[0];
+
+  tree.write(moduleFederationConfigPath, printer.printFile(updatedSourceCode));
+}
+
+/**
+ * TODO (nicholas): remove this before merge
+ * @param tree The workspace tree
+ * @param remoteName The name of the remote to check
+ * @returns Remote ProjectConfig if it exists, false otherwise
+ */
+export async function checkRemoteExists(tree: Tree, remoteName: string) {
+  const remote = await getRemote(remoteName);
+  if (!remote) return false;
+  let hasModuleFederationConfig = false;
+  if (tree.exists(join(remote.root, 'module-federation.config.ts'))) {
```

Comment:
We should use `joinPathFragments` instead of `join` in generator code to account for windows. Our virtual FS uses unix style paths, but on windows, join will create windows style paths.